django/university/routes/exchange/DirectExchangeView.py

```diff
@@ -0,0 +1,98 @@
+
+class DirectExchangeView(View):
+    def get(self, request):
+        return HttpResponse()
+
+    def post(self, request):
+        student_schedules = {}
+        
+        sigarra_res = SigarraController().get_student_schedule(request.user.username)
+        
+        if (sigarra_res.status_code != 200):
+            return HttpResponse(status=sigarra_res.status_code)
+
+        username = request.user.username
+        schedule_data = sigarra_res.data
+        
+        student_schedules[username] = build_student_schedule_dict(schedule_data)
+
+        exchange_choices = request.POST.getlist('exchangeChoices[]')
+        exchanges = list(map(lambda exchange : json.loads(exchange), exchange_choices))
+
+        # Add the other students schedule to the dictionary
+        (status, trailing) = build_student_schedule_dicts(student_schedules, exchanges)
+        if status == ExchangeStatus.FETCH_SCHEDULE_ERROR:
+            return HttpResponse(status=trailing)
+
+        for student in student_schedules.keys():
+            student_schedule = list(student_schedules[student].values())
+            update_schedule_accepted_exchanges(student, student_schedule)
+            student_schedules[student] = build_student_schedule_dict(student_schedule)
+
+        exchange_model = DirectExchange(accepted=False, issuer_name=f"{request.user.first_name} {request.user.last_name}", issuer_nmec=request.user.username)
+
+        (status, trailing) = build_new_schedules(
+            student_schedules, exchanges, request.user.username)
+        
+        if status == ExchangeStatus.STUDENTS_NOT_ENROLLED:
+            return JsonResponse({"error": incorrect_class_error()}, status=400, safe=False)
+    
+        inserted_exchanges = []
+        (status, trailing) = ExchangeController.create_direct_exchange_participants(student_schedules, exchanges, inserted_exchanges, exchange_model, request.user.username)
+        
+        if status == ExchangeStatus.CLASSES_OVERLAP:    
+            return JsonResponse({"error": "classes-overlap"}, status=400, safe=False)
+
+        exchange_model.save()
+    
+        tokens_to_generate = {}
+        for inserted_exchange in inserted_exchanges:
+            participant = inserted_exchange.participant;
+            if not(participant in tokens_to_generate):
+                token = jwt.encode({"username": participant, "exchange_id": exchange_model.id, "exp": (datetime.datetime.now() + datetime.timedelta(seconds=VERIFY_EXCHANGE_TOKEN_EXPIRATION_SECONDS)).timestamp()}, JWT_KEY, algorithm="HS256")
+                tokens_to_generate[participant] = token
+                html_message = render_to_string('confirm_exchange.html', {'confirm_link': f"{DOMAIN}tts/verify_direct_exchange/{token}"})
+                send_mail(
+                    'Confirmação de troca',
+                    strip_tags(html_message),
+                    'tts@exchange.com',
+                    [f'up{participant}@up.pt']
```

Comment:
I think this email should be in env, and we should discuss what it may be.

Reply:
You are referring to the `tts@exchange.com` email, right?

Reply:
yes!
But also, the student may be tested. I don't know with `edu.<>.up.pt` work better or not

Reply:
It is now done